### PR TITLE
VAULT-18284: Audit refactor packages

### PR DESCRIPTION
**Pull request by peteski22**
This PR refactors a lot of the `event` package which resides in `internal/observability` into the `audit` package.

The reason for this is to prevent circular dependencies in packages which need to use `event` (of which `audit` is one), and to begin to remove domain specific code from the generic package.

Files were moved using `git mv` but not all seem to have managed to remember that happened to them in the PR (some show moved, some seem to indicate they're new/deleted) 😞. Comments have been added to give some clarity on why things look like 'a lot'.

A future PR will be created to move structs and interfaces into a `types.go` file in the `audit` package to group them together as currently not everything sits somewhere sensible.

### CODE REVIEW

File: audit/audit.go
Comment (peteski22) [inline audit/audit.go:1]: `audit.go` was moved to `event.go`

File: internal/observability/event/node_formatter_audit_json_test.go
Comment (peteski22) [inline internal/observability/event/node_formatter_audit_json_test.go:1]: Moved from `internal/observability/event/node_formatter_audit_json_test.go` to `audit/entry_formatter_test.go`

File: audit/format.go
Comment (peteski22) [inline audit/format.go:1]: Moved contents to other files:

Types will be moved to a `types.go` in a future PR to group structs and interfaces together.

`audit/entry_formatter_writer.go`

Interfaces:  

* `Salter`
* `Writer`
* `Formatter`

Structs:

* `AuditFormatter` => `EntryFormatter`
* `AuditFormatterWriter` => `EntryFormatterWriter`
* `FormatterConfig`
* `nonPersistentSalt`

Request/Response related types and methods have moved to: `audit/entry_formatter.go`

File: internal/observability/event/audit_sink_file.go
Comment (peteski22) [inline internal/observability/event/audit_sink_file.go:1]: Moved from `internal/observability/event/audit_sink_file.go `to `internal/observability/event/sink_file.go`

File: internal/observability/event/audit_sink_file_test.go
Comment (peteski22) [inline internal/observability/event/audit_sink_file_test.go:1]: Moved from `internal/observability/event/audit_sink_file_test.go` to `internal/observability/event/sink_file_test.go`

TL;DR:

`audit_sink_file_test` => `sink_file_test`

File: internal/observability/event/node_formatter_audit_jsonx.go
Comment (peteski22) [inline internal/observability/event/node_formatter_audit_jsonx.go:1]: Merged JSON + JSONx formatter nodes which can now be a found in `audit/entry_formatter.go`

File: internal/observability/event/sink_file.go
Comment (peteski22) [inline internal/observability/event/sink_file.go:1]: Moved from `audit_sink_file`.

File: internal/observability/event/sink_file_test.go
Comment (peteski22) [inline internal/observability/event/sink_file_test.go:1]: Moved from `audit_sink_file_test`.

**Review by marcboudreau — APPROVED** [review]
LGTM
